test/e2e/validation_test.go

```diff
@@ -165,37 +154,50 @@ func validateInstanceSecret(kubeclient kubernetes.Interface, namespace, instance
 			return false, err
 		}
 		encodedCreds := instanceSecret.Data[instanceID]
-
-		var creds tracker.Credentials
 		if err := json.Unmarshal(encodedCreds, &creds); err != nil {
 			return false, errors.Wrap(err, "unmarshalling creds failed")
 		}
-
-		if err := validateConnectivity(ipAddress, instanceID, creds); err == nil {
-			log.Println("Successfully validated the SSH Connection")
-			return true, nil
-		}
-
-		log.Printf("failed with error for creds %v: %v", creds, err)
-		return false, nil
+		return true, nil
 	})
+	return creds, err
+}
+
+// validateInstanceSecret validates the instance secret.
+func (tc *testContext) validateInstanceSecret(instanceID string) error {
+	ipAddress, err := tc.getInstanceIP(instanceID)
+	if err != nil {
+		return err
+	}
+	creds, err := tc.getCredsSecretData(instanceID)
+	if err != nil {
+		return err
+	}
+	if creds == (tracker.Credentials{}) {
+		return errors.New("expected credentials to be present but got a nil value")
+	}
+	windowsVM, err := tc.getWindowsVM(ipAddress, instanceID, creds)
+	if err != nil {
+		return err
+	}
+	gc.windowsVMs = append(gc.windowsVMs, windowsVM)
+	err = tc.validateConnectivity(windowsVM)
+	if err == nil {
+		log.Println("Successfully established ssh connection")
```

Comment:
Either you remove this or provide some context in the log message. Otherwise someone has to remember to add context when we increase the number of nodes.

Reply:
Removed. Anyways, this entire function need to be removed +1.